CleverTapSDK/InApps/CTInAppStore.m

```diff
@@ -196,6 +209,127 @@ - (NSDictionary *)dequeueInApp {
     }
 }
 
+#pragma mark Delayed InApps
+
+- (void)migrateDelayedInAppQueueKeys {
+    @synchronized(self) {
+        NSString *storageKey = [CTPreferences storageKeyWithSuffix:CLTAP_PREFS_DELAYED_INAPP_KEY config: self.config];
+        id data = [CTPreferences getObjectForKey:storageKey];
+        if (data) {
+            if ([data isKindOfClass:[NSArray class]]) {
+                _delayedInAppsQueue = data;
+                
+                NSString *encryptedString = nil;
+                @try {
+                    encryptedString = [self.cryptManager encryptObject:data];
+                    if (!encryptedString) {
+                        CleverTapLogInternal(self.config.logLevel, @"%@: Encryption failed", self);
+                        return;
+                    }
+                } @catch (NSException *exception) {
+                    CleverTapLogInternal(self.config.logLevel, @"%@: Encryption error: %@", self, exception);
+                    return;
+                }
+                
+                NSString *newStorageKey = [self storageKeyWithSuffix:CLTAP_PREFS_DELAYED_INAPP_KEY];
+                [CTPreferences putString:encryptedString forKey:newStorageKey];
+            }
+            [CTPreferences removeObjectForKey:storageKey];
+        }
+    }
+}
+
+- (void)storeDelayedInApps:(NSArray *)inApps {
+    if (!inApps) return;
+    
+    @synchronized (self) {
+        _delayedInAppsQueue = inApps;
+        
+        NSString *encryptedString = [self.cryptManager encryptObject:inApps];
+        NSString *storageKey = [self storageKeyWithSuffix:CLTAP_PREFS_DELAYED_INAPP_KEY];
+        [CTPreferences putString:encryptedString forKey:storageKey];
+    }
+}
+
+- (NSArray *)delayedInAppsQueue {
+    @synchronized(self) {
+        if (_delayedInAppsQueue) return _delayedInAppsQueue;
+        
+        NSString *storageKey = [self storageKeyWithSuffix:CLTAP_PREFS_DELAYED_INAPP_KEY];
+        id data = [CTPreferences getObjectForKey:storageKey];
+        
```

Comment:
Can we initialise _delayedInAppsQueue = [NSArray new] here so that we have not to update at every places, and remove at other places